### ISSUE TITLE
sample rows from multi_PRS in SCT

### ISSUE THREAD

**Issue by gaochengPRC** [privefl/bigstatsr#110]
My multi_PRS is 10000 * 5600, that is, training sample size is 10000, 5600  polygenic risk scores. Now I want to sample 10000 rows with replacement from multi_PRS, I found that neither `snp_grid_stacking(multi_PRS[sample(1:10000, 10000, replace = TRUE),] , y[ind.train], ncores = nb_cores())` 
nor  
`snp_grid_stacking(as_FBM(multi_PRS[sample(1:10000, 10000, replace = TRUE),] ), y[ind.train], ncores = nb_cores())` 
worked. My idea is 
`ind.sample <- sort(sample(1:l, l, replace = TRUE)) # l <- dim(G.train)[1]`
`G.train.b <- big_copy(G.train, ind.row = ind.sample)`
`multi_PRS_b <- snp_grid_PRS(
    G.train.b, all_keep, betas = beta_gwas, lpS = lpval,
    n_thr_lpS = 50, ncores = nb_cores())`

Since the genomic data is quite large, to save computational time, do you have a better idea to deal with this? Thank you very much!

**Comment by privefl** [comment]
You can use parameter `ind.train = sample(10000, replace = TRUE)` in `snp_grid_stacking()` to provide the row indices from `multi_PRS` that you want to use. It's part of the parameters you can pass to `big_spL**Reg()` through `...`.

Note that as `big_spL**Reg()` is performing cross-validation internally, so it is not recommended to use sampling with replacement (as some indices from the training sets will be in the validation sets). You could define the fold indices so that they are not part of multiple folds if you really want to do that.

**Comment by gaochengPRC** [comment]
So, it makes sense to work with my code instead of `snp_grid_stacking(multi_PRS[sample(10000, replace = TRUE),] , y[ind.train], ncores = nb_cores())`. In addition, the code `snp_grid_stacking(multi_PRS[sample(10000, replace = TRUE),] , ......)` doesn't work. In my case, it reported errors, I tried many times. I also tried `snp_grid_stacking(as_FBM(multi_PRS[sample(10000, replace = TRUE),]), ......)`, still  failed. So, I feel quite confused with the errors.

**Comment by privefl** [comment]
You need to provide `ind.train = sample(10000, replace = TRUE)` as a parameter.
And provide the `y.train` that corresponds to `ind.train`.
